internal/fsutil/create.go

```diff
@@ -18,8 +21,29 @@ type CreateOptions struct {
 	MakeParents bool
 }
 
+type Info struct {
+	Path string
+	Mode fs.FileMode
+	Hash string
+	Size uint
+	Link string
+}
+
+type Creator struct {
+	// Created keeps track of information about the files created. If a file
+	// is created several times it only tracks the latest one.
+	Created map[string]Info
+}
+
+func NewCreator() *Creator {
+	return &Creator{Created: make(map[string]Info)}
+}
+
 // Creates a filesystem entry according to the provided options.
-func Create(o *CreateOptions) error {
+func (c Creator) Create(o *CreateOptions) error {
```

Comment:
The self parameter should be passed as a pointer instead of by value, since the type has identity properties and we don't want to be copying it over on method calls.